lib/src/main/kotlin/com/swmansion/starknet/service/http/requests/HttpBatchRequest.kt

```diff
@@ -0,0 +1,65 @@
+package com.swmansion.starknet.service.http.requests
+
+import com.swmansion.starknet.provider.Request
+import com.swmansion.starknet.provider.rpc.JsonRpcRequest
+import com.swmansion.starknet.provider.rpc.buildJsonHttpBatchDeserializer
+import com.swmansion.starknet.provider.rpc.buildJsonHttpBatchDeserializerOfDifferentTypes
+import com.swmansion.starknet.service.http.HttpService
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import java.util.concurrent.CompletableFuture
+
+class HttpBatchRequest<T> private constructor(
```

Comment:
Why not 
```suggestion
class HttpBatchRequest<HttpBatchRequestType> private constructor(
```
or 
```suggestion
class HttpBatchRequest<T: HttpBatchRequestType> private constructor(
```

This applies not here specifically, but to every other places where `<T>` is still used. 

I think we might want to use some of that additional type safety that comes with introduction of `HttpBatchRequestType`, WDYT?

Reply:
Current implementation keeps type safety for `batchRequests` and allows to use `batchRequestsAny` with all types which implement `HttpBatchRequestType`.  Why should we change it for some of the suggested solutions?

> I think we might want to use some of that additional type safety

Do you have something specific on your mind? As mentioned before, currently all types which implement `HttpBatchRequestType` can ba used in `batchRequestsAny`.

Reply:
I applied second suggestion in all occurences 👍 

> I think we might want to use some of that additional type safety

Do you have something specific on your mind? Currently `batchRequests` allows to send requests of only one type, `batchRequestsAny` allows to use all types which implement `HttpBatchRequestType`